File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        j = count = ans = 0
        for i, num in enumerate(nums):
            if num & 1:
                k -= 1
                count = 0
            while k == 0:
                k += nums[j] & 1
                j += 1
                count += 1
            ans += count
        return ans

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numberOfSubarrays = function(nums, k) {
  let [j, count, ans] = [0, 0, 0];
  
  nums.forEach((num, i) => {
    if (num & 1) {
      k--;
      count = 0;
    }
    
    while (!k) {
      k += nums[j++] & 1;
      count++;
    }
    
    ans += count;
  })
  
  return ans;
};